### PR TITLE
CLN: flake8 cleanup of core.internals

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/internals.py [hunk 1]
```diff
@@ -1108,7 +1108,8 @@ def check_int_bool(self, inplace):
         # a fill na type method
         try:
             m = missing.clean_fill_method(method)
-        except:
+        except Exception:
```
Comment (jreback) [inline pandas/core/internals.py]: the entire point of correctly lining is to catch a more specific error
don’t put Execption everywhere with TODO that is not useful
his is why we are not raising on master atm with the latest linter yet

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals.py (hunk 1 above)
Comment (TomAugspurger) [reply pandas/core/internals.py]: Do you know the appropriate exceptions to catch? I was going to fix these but didn't know what could be raised, and didn't want to break things.

Comment (jreback) [reply pandas/core/internals.py]: you simply put a breakpoint in the exception and run the tests

**Comment by TomAugspurger** [comment]
How confident are you that we have tests exercising all the exceptions that
get raised?

**Comment by jreback** [comment]
its likeley to be ``TypeError, ValueError``, but I'd rather error on the side of havining a bug pop up rather than keep hiding it (like we are doing now)

File: pandas/core/internals.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals.py]: Well looking at clean_fill_method the only thing it'll raise is ValueError, so this one I can make specific.

Comment (jbrockmendel) [reply pandas/core/internals.py]: Ditto for clean_interp_method.

**Comment by jbrockmendel** [comment]
The only one left is

```
        try:
            return self.sp_index.length
        except:
            return 0
```

Any idea what the failure mode here is?

File: pandas/core/internals.py [hunk 2]
```diff
@@ -840,7 +840,7 @@ def setitem(self, indexer, value, mgr=None):
 
         transf = (lambda x: x.T) if self.ndim == 2 else (lambda x: x)
         values = transf(values)
-        l = len(values)
+        vlen = len(values)
```
Comment (jschendel) [inline pandas/core/internals.py]: It looks like this is only used in one place (line 858).  Is there a reason to make it a variable in this case?  Seems like we could just replace the one instance where `vlen` is used with `len(values)`.

Comment (jbrockmendel) [reply pandas/core/internals.py]: Sounds good, will change.

File: pandas/core/internals.py [hunk 3]
```diff
@@ -2807,7 +2806,8 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
     def __len__(self):
         try:
             return self.sp_index.length
-        except:
+        except Exception:
```
Comment (jreback) [inline pandas/core/internals.py]: enumerate these

**Review by jreback — CHANGES_REQUESTED** [review]
pls enumerate all of the exceptions rather than Exception

File: pandas/core/internals.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/internals.py]: Please read my previous comment.

Comment (jreback) [reply pandas/core/internals.py]: I did it was not helpful. you need to specify here.

Comment (jbrockmendel) [reply pandas/core/internals.py]: > I did it was not helpful. you need to specify here.

Of course it wasn't helpful.  It was asking for help!

Screw it, closing.

Comment (TomAugspurger) [reply pandas/core/internals.py]: > Any idea what the failure mode here is?

@jreback was the comment @jbrockmendel referred to. Saying it was "not helpful" is driving away contributors.

Comment (jreback) [reply pandas/core/internals.py]: @jbrockmendel my point is that saying ping I am ready when you haven't answer my questions is not helpful.

Comment (jreback) [reply pandas/core/internals.py]: I spend an enormous amounts of my time reviewing PR's and we have a lot of them. I do except answers to comments that I make.

Comment (jbrockmendel) [reply pandas/core/internals.py]: I'm guessing these last two comments are about a "ping" over in #19139, where I _had_ answered the question but that answer was ignored.

**Pull request closed without merging**